core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -192,29 +185,6 @@ public static PositionDeleteIndex toPositionIndex(CloseableIterable<Long> posDel
     }
   }
 
-  public static <T> CloseableIterable<T> streamingFilter(
-      CloseableIterable<T> rows,
-      Function<T, Long> rowToPosition,
-      CloseableIterable<Long> posDeletes) {
-    return streamingFilter(rows, rowToPosition, posDeletes, new DeleteCounter());
-  }
-
-  public static <T> CloseableIterable<T> streamingFilter(
-      CloseableIterable<T> rows,
-      Function<T, Long> rowToPosition,
-      CloseableIterable<Long> posDeletes,
-      DeleteCounter counter) {
-    return new PositionStreamDeleteFilter<>(rows, rowToPosition, posDeletes, counter);
-  }
-
-  public static <T> CloseableIterable<T> streamingMarker(
-      CloseableIterable<T> rows,
-      Function<T, Long> rowToPosition,
-      CloseableIterable<Long> posDeletes,
-      Consumer<T> markDeleted) {
-    return new PositionStreamDeleteMarker<>(rows, rowToPosition, posDeletes, markDeleted);
-  }
-
```

Comment:
I'm all for removing unneccessary APIs but since they're public I think this should go through a deprecation process.

Reply:
I'd argue that `Deletes` is actually a utility class with just utility functions (static methods), and the reason it is public is that the functions are called from another module (data). As such, I think it is a bit different.

However, if you insist, I'll deprecate the static methods for now.